### PR TITLE
feat(misconf): Added fs.FS based scanning via latest defsec

### PR DESCRIPTION
**Pull request by liamg**
## Description

This adds io/fs.FS based scanning to Trivy, as described [here](https://github.com/aquasecurity/fanal/pull/460/commits/f94f4d266e6aeccfaf44db17e5bd4fdac92cee76).

It enables misconfiguration detection in container image scanning as well.

### CODE REVIEW

**Review by knqyf263 — COMMENTED** [review]
Looks like your formatting is different from gofmt as they use tabs for indentation.
https://go.dev/doc/effective_go#formatting

I saw many diffs regarding tabs vs spaces in this PR. Am I missing something?

**Comment by liamg** [comment]
I probably forgot to enable gofmt/goimports in my editor settings for the project. I'll sort it out once I come back to this when the fanal PR is merged :+1:

**Comment by knqyf263** [comment]
Thanks. I'm reviewing https://github.com/aquasecurity/fanal/pull/460 now.

**Comment by knqyf263** [comment]
```
$ trivy conf -f json -o result.json ./
$ jq . result.json
...
        {
          "Type": "Kubernetes Security Check",
          "ID": "KSV023",
          "Title": "hostPath volumes mounted",
          "Message": "Deployment 'hello-kubernetes' should not set 'spec.template.volumes.hostPath'",
          "Namespace": "appshield.kubernetes.KSV023",
          "Query": "data.appshield.kubernetes.KSV023.deny",
          "Resolution": "Do not set 'spec.volumes[*].hostPath'.",
          "Severity": "MEDIUM",
          "PrimaryURL": "https://avd.aquasec.com/appshield/ksv023",
          "References": [
            "https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline",
            "https://avd.aquasec.com/appshield/ksv023"
          ],
          "Status": "FAIL",
          "Layer": {},
          "IacMetadata": {
            "Provider": "Kubernetes",
            "Service": "general"
          }
        },
```

I still didn't see line numbers in Kubernetes checks. Do we need additional implementation?

**Comment by knqyf263** [comment]
One more thing is that Trivy assumes capital letters in severity, but DefSec may have camel case in severity. We can convert it somewhere or have all capital severities in DefSec.

```
$ ./trivy conf .
2022-05-06T19:22:28.905+0300    INFO    Detected config files: 22
2022-05-06T19:22:28.905+0300    WARN    severity must be [UNKNOWN LOW MEDIUM HIGH CRITICAL], but Critical
```

**Comment by knqyf263** [comment]
Also, looks like the misconf results are duplicated.

```
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -67,6 +67,6 @@
        	            	     Successes: (int) 0,
        	            	-    Failures: (int) 1,
        	            	+    Failures: (int) 2,
        	            	     Exceptions: (int) 0
        	            	    }),
        	            	-   Misconfigurations: ([]types.DetectedMisconfiguration) (len=1) {
        	            	+   Misconfigurations: ([]types.DetectedMisconfiguration) (len=2) {
        	            	     (types.DetectedMisconfiguration) {
        	            	@@ -75,3 +75,3 @@
```

Could you double check by running the following test?
https://github.com/aquasecurity/trivy/blob/8a4b49ca7065ade76524c7bbbf9bfda53ed270a4/integration/fs_test.go#L67-L75

File: integration/fs_test.go [hunk 1]
```diff
@@ -167,6 +169,9 @@ func TestFilesystem(t *testing.T) {
 			// Run "trivy fs"
 			assert.Nil(t, app.Run(osArgs))
 
+			data, _ := ioutil.ReadFile(outputFile)
+			fmt.Println(string(data))
```
Comment (knqyf263) [inline integration/fs_test.go:173]: maybe for testing?

File: pkg/scanner/local/scan.go [hunk 2]
```diff
@@ -1,362 +1,362 @@
...
-	ftypes "github.com/aquasecurity/fanal/types"
-	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
-	"github.com/aquasecurity/trivy/pkg/detector/library"
-	ospkgDetector "github.com/aquasecurity/trivy/pkg/detector/ospkg"
-	"github.com/aquasecurity/trivy/pkg/log"
-	"github.com/aquasecurity/trivy/pkg/types"
-
-	_ "github.com/aquasecurity/fanal/analyzer/all"
-	_ "github.com/aquasecurity/fanal/hook/all"
+    "errors"
```
Comment (knqyf263) [inline pkg/scanner/local/scan.go:4]: tabs vs spaces

**Comment by knqyf263** [comment]
It cannot be built for FreeBSD.
https://github.com/aquasecurity/trivy/runs/6358741081?check_suite_focus=true

I guess squealer doesn't support FreeBSD. Can we add support for it?
https://github.com/owenrumney/squealer/blob/3d5ba40fea05b5ecc8b299bbe56ec86cf462a1d1/internal/pkg/scan/signals_unix.go

File: pkg/scanner/local/scan.go (hunk 2 above)
Comment (liamg) [reply pkg/scanner/local/scan.go]: Sorry, forgot about this!

**Comment by liamg** [comment]
@owenrumney has added FreeBSD support now - here is the fanal PR: https://github.com/aquasecurity/fanal/pull/520

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**